power-control-x86/src/power_control.cpp

```diff
@@ -784,16 +784,26 @@ static int setGPIOOutputForMs(const std::string& name, const int value,
 static void powerOn()
 {
     setGPIOOutputForMs("ASSERT_PWR_BTN_L", 0, powerPulseTimeMs);
+
+    //Enable yaap service only the first time
```

Comment:
This block code should be moved before setGPIOOutputForMs("ASSERT_PWR_BTN_L", 0, powerPulseTimeMs); so that there are no race conditions.